neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/dialect/impl/Neo4jAssociationQueries.java

```diff
@@ -252,4 +275,343 @@ private EntityKey getEntityKey(AssociationKey associationKey, RowKey rowKey) {
 	private static void appendRelationshipType(StringBuilder queryBuilder, AssociationKeyMetadata associationKeyMetadata) {
 		escapeIdentifier( queryBuilder, associationKeyMetadata.getCollectionRole() );
 	}
+
+	/**
+	 * Give an embedded association, creates all the nodes and relationships required to represent it.
+	 * It assumes that the entity node containing the association already exists in the db.
+	 *
+	 * @param executionEngine the {@link ExecutionEngine} to run the query
+	 * @param associationKey the {@link AssociationKey} identifying the association
+	 * @param embeddedKey the {@link EntityKey} identifying the embedded component
+	 * @return the created {@link Relationship} that represents the association
+	 */
+	public Relationship createRelationshipForEmbeddedAssociation(ExecutionEngine executionEngine, AssociationKey associationKey, EntityKey embeddedKey) {
+		String query = initCreateEmbeddedAssociationdQuery( associationKey, embeddedKey );
+		Object[] queryValues = createRelationshipForEmbeddedQueryValues( associationKey, embeddedKey );
+		return executeQuery( executionEngine, query, queryValues );
+	}
+
+	private String initCreateEmbeddedAssociationdQuery(AssociationKey associationKey, EntityKey embeddedKey) {
```

Comment:
The query might be different if some values are null because it might not need to create intermediate nodes.
I'll find an example.